lib/pages/chat/events/video_player.dart

```diff
@@ -1,160 +1,177 @@
+              if (hasThumbnail)
+                Center(
+                  child: ImageBubble(
+                    widget.event,
+                    tapToView: false,
+                    width: MessageContentStyle.imageBubbleWidth(width),
+                    height: MessageContentStyle.videoBubbleHeight(height),
                   ),
                 )
-              : Stack(
-                  children: [
-                    if (hasThumbnail)
-                      Center(
-                        child: ImageBubble(
-                          widget.event,
-                          tapToView: false,
-                        ),
-                      )
-                    else
-                      BlurHash(hash: blurHash),
-                    Center(
-                      child: OutlinedButton.icon(
-                        style: OutlinedButton.styleFrom(
-                          backgroundColor:
-                              Theme.of(context).colorScheme.surface,
-                        ),
-                        icon: _isDownloading
-                            ? const SizedBox(
-                                width: 24,
-                                height: 24,
-                                child: CircularProgressIndicator.adaptive(
-                                  strokeWidth: 2,
-                                ),
-                              )
-                            : const Icon(Icons.download_outlined),
-                        label: Text(
-                          _isDownloading
-                              ? L10n.of(context)!.loadingPleaseWait
-                              : L10n.of(context)!.videoWithSize(
-                                  widget.event.sizeString ?? '?MB',
-                                ),
-                        ),
-                        onPressed: _isDownloading ? null : _downloadAction,
-                      ),
-                    ),
-                  ],
+              else
+                BlurHash(hash: blurHash),
+              Center(
+                child: ValueListenableBuilder<DownloadVideoState>(
+                  valueListenable: _downloadStateNotifier,
+                  builder: (context, downloadState, child) {
+                    switch (downloadState) {
+                      case DownloadVideoState.loading:
+                        return Stack(
+                          alignment: Alignment.center,
+                          children: [
+                            _CenterVideoButton(
+                              icon: Icons.play_arrow,
+                              onTap: _downloadAction,
+                            ),
+                            SizedBox(
+                              width: MessageContentStyle.videoCenterButtonSize,
+                              height: MessageContentStyle.videoCenterButtonSize,
+                              child: CircularProgressIndicator(
+                                strokeWidth: 2,
```

Comment:
Reuse?
![Uploading Screenshot 2023-09-22 at 14.07.45.png…]()

Reply:
reuse what ? this is only for video

Reply:
Reuse CircularProgressIndicator.

Reply:
the CircularProgressIndicator is used for making the loading around play button, if only use for video, so it not for reused